File: examples/python/number/add.py
"""
Date Created:               2018-02-27
Date Modified:              2018-02-27
Version:                    1
Contract Hash:              28dfdb9d1d0fa0b86804e4a4e0f0c42484f32792
Available on NEO TestNet:   False
Available on CoZ TestNet:   False
Available on MainNet:       False

Example:
    Test Invoke:            build /path/to/add.py test 0202 02 False False 2 5
    Expected Result:        7
    Operation Count:        11
    GAS Consumption:        0.016
"""


def Main(a, b):
    """
    :param a: First input number of concern
    :param b: Second input number of concern
    :type a: int
    :type b: int
    :return: Sum of the 2 input numbers
    :rtype: int
    """
    c = a + b  # It is a good practice to always allocate calculated value to a local variable instead of chaining to return or another function
    return c

File: examples/csharp/number/add.cs
using Neo.SmartContract.Framework;

// Date Created:                2018-03-03
// Date Modified:               2018-03-93
// Version:                     1
// Contract Hash:               0x4f088dab7a5167adeda956881c213ace55b64ebd
// Available on NEO TestNet:    False
// Available on CoZ TestNet:    False
// Available on MainNet:        False

// Example  
// Input Params     :           0202
// Output Params    :           02
// Expected Result  :           7
// Operation Count  :           Contract compiled in Debug: 60 / Contract compiled in Release: 34
// GAS Consumption  :           Contract compiled in Debug:     0.043 (NEO Debugger) / 0.054 (neo-gui-developer)  
// GAS Consumption  :           Contract compiled in Release:   0.024 (NEO Debugger) / 0.035 (neo-gui-developer)

namespace SumSmartContract
{
    public class Contract : SmartContract
    {
        public static int Main(int a, int b)
        {
            var c = a + b;
            return c;
        }
    }
}